corehq/apps/app_manager/util.py

```diff
@@ -431,9 +431,13 @@ def get_session_schema(form):
     from corehq.apps.app_manager.suite_xml.sections.entries import EntriesHelper
     structure = {}
     datums = EntriesHelper(form.get_app()).get_datums_meta_for_form_generic(form)
-    datums = [d for d in datums if not d.is_new_case_id and d.case_type and d.datum.id == 'case_id']
+    datums = [
+        d for d in datums
+        if not d.is_new_case_id and d.case_type and d.requires_selection
+    ]
     if len(datums):
-        structure['case_id'] = {
+        session_var = datums[-1].datum.id
+        structure[session_var] = {
```

Comment:
@millerdev yes the session var is sometimes something other than `case_id` e.g. in child modules. That's what this change is all about.

I have tested this locally and it works fine. 

Re the docs - they leave a lot unexplained e.g. what is `"key": "@case_type"` used for (the case type isn't even included in the schema anywhere except the name)? Do the subset ID's mean anything i.e. if I change `parent` to `mother` will it break things? How do relationships work - are the relationship names important?